### PR TITLE
Add warning and update wording for extended statistics

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

- Adds a warning and update wording, to align with the refined Chinese version https://github.com/pingcap/docs-cn/pull/17339
- The word "create" is more commonly used together with "extended statistics objects". So I also updated the word usage.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/17339, https://github.com/pingcap/docs/pull/9891, https://docs.pingcap.com/tidb/dev/extended-statistics

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: extended-statistics.md [hunk 1]
```diff
@@ -5,20 +5,18 @@ summary: Learn how to use extended statistics to guide the optimizer.
...
 - Extended statistics: statistics filtered by tables and columns.
 
-> **Tip:**
->
-> Before reading this document, it is recommended that you read [Introduction to Statistics](/statistics.md) first.
-
 When the `ANALYZE` statement is executed manually or automatically, TiDB by default only collects the basic statistics and does not collect the extended statistics. This is because the extended statistics are only used for optimizer estimates in specific scenarios, and collecting them requires additional overhead.
 
-Extended statistics are disabled by default. To collect extended statistics, you need to first enable the extended statistics, and then register each individual extended statistics object.
+Extended statistics are disabled by default. To collect extended statistics, you need to first enable the extended statistics, and then create every single extended statistics object. After the objects have been created, the next time the `ANALYZE` statement is executed, TiDB collects both the basic statistics and the corresponding extended statistics.
```
Comment (lilin90) [inline extended-statistics.md:15]:
```suggestion
Extended statistics are disabled by default. To collect extended statistics, you need to first enable the extended statistics, and then create every single extended statistics object. After the objects have been created, the next time the `ANALYZE` statement is executed, TiDB collects both the basic statistics and the corresponding extended statistics of the created objects.
```

File: extended-statistics.md [hunk 2]
```diff
@@ -5,20 +5,18 @@ summary: Learn how to use extended statistics to guide the optimizer.
 
 # Introduction to Extended Statistics
 
-TiDB can collect the following two types of statistics:
+TiDB can collect the following two types of statistics. This documents describes how to use extended statistics to guide the optimizer. Before reading this document, it is recommended that you read [Introduction to Statistics](/statistics.md) first.
 
 - Basic statistics: statistics such as histograms and Count-Min Sketch. See [Introduction to Statistics](/statistics.md) for details.
 - Extended statistics: statistics filtered by tables and columns.
```
[the remark this answers is not in the document]
Comment (lilin90) [reply extended-statistics.md:11]: @qiancai Thanks for trying to add detailed explanation. @winoros PTAL, thanks!

**Review by winoros — APPROVED** [review]

File: extended-statistics.md [hunk 3]
```diff
@@ -5,20 +5,18 @@ summary: Learn how to use extended statistics to guide the optimizer.
...
+TiDB can collect the following two types of statistics. This documents describes how to use extended statistics to guide the optimizer. Before reading this document, it is recommended that you read [Introduction to Statistics](/statistics.md) first.
 
-- Basic statistics: statistics such as histograms and Count-Min Sketch. See [Introduction to Statistics](/statistics.md) for details.
-- Extended statistics: statistics filtered by tables and columns.
-
-> **Tip:**
->
-> Before reading this document, it is recommended that you read [Introduction to Statistics](/statistics.md) first.
+- Basic statistics: statistics such as histograms and Count-Min Sketch, which primarily focus on individual columns. They are essential for the optimizer to estimate the query cost. See [Introduction to Statistics](/statistics.md) for details.
+- Extended statistics: statistics that focus on data correlations between specified columns, which guide the optimizer to estimate the query cost more precisely when the queried columns are correlated.
```
Comment (lilin90) [inline extended-statistics.md:11]: Note: I'll update the corresponding two lines for docs in Chinese in a coming PR.

File: extended-statistics.md [hunk 4]
```diff
@@ -5,20 +5,18 @@ summary: Learn how to use extended statistics to guide the optimizer.
 
 # Introduction to Extended Statistics
 
-TiDB can collect the following two types of statistics:
+TiDB can collect the following two types of statistics. This documents describes how to use extended statistics to guide the optimizer. Before reading this document, it is recommended that you read [Introduction to Statistics](/statistics.md) first.
```
Comment (qiancai) [inline extended-statistics.md:8]:
```suggestion
TiDB can collect the following two types of statistics. This document describes how to use extended statistics to guide the optimizer. Before reading this document, it is recommended that you read [Introduction to Statistics](/statistics.md) first.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**